cli/src/commands/parse.rs

```diff
@@ -17,6 +17,19 @@ pub struct ParsedLockfile {
     pub packages: Vec<PackageDescriptor>,
 }
 
+impl From<ParsedLockfile> for Vec<PackageDescriptorAndLockfilePath> {
+    fn from(value: ParsedLockfile) -> Self {
+        let mut packages = Vec::with_capacity(value.packages.len());
+        for pkg_descriptor in value.packages {
+            packages.push(PackageDescriptorAndLockfilePath {
+                package_descriptor: pkg_descriptor,
+                lockfile_path: Some(value.path.to_string_lossy().to_string()),
```

Comment:
`Cow::into_owned` is more efficient here (because `.to_string()` always copies while `.into_owned()` only copies if needed).

```suggestion
                lockfile_path: Some(value.path.to_string_lossy().into_owned()),
```